rmm/src/rmi/realm/params.rs

```diff
@@ -129,6 +129,10 @@ impl Params {
         let ipa_bits = self.ipa_bits();
         let rtt_slvl = self.rtt_level_start as usize;
 
+        if rtt_slvl > RTT_PAGE_LEVEL {
+            return Err(Error::RmiErrorInput);
+        }
+
         let level = RTT_PAGE_LEVEL - rtt_slvl;
```

Comment:
Nice work! Just a minor suggestion: How about using `checked_sub` to make it more concise?

```suggestion
let level = RTT_PAGE_LEVEL.checked_sub(rtt_slvl).ok_or(Error::RmiErrorInput)?;
```